com.playeveryware.eos/Runtime/Core/EOSManager.cs

```diff
@@ -138,6 +135,40 @@ public delegate void OnAuthLinkExternalAccountCallback(
         //cached log levels for retrieving later
         private static Dictionary<LogCategory, LogLevel> logLevels;
 
+        /// <summary>
+        /// Lazy-loaded singleton reference for the EOSManager.
+        /// If it doesn't exist when queried, it will create a GameObject,
+        /// and add an EOSManager component to it.
+        /// This can serve as an alternate way of setting up EOSManager, rather
+        /// than using a prefab with the script on it.
+        /// </summary>
+        private static Lazy<EOSManager> s_lazyManagerInstance = new Lazy<EOSManager>(() =>
+        {
+            EOSSingleton.print($"{nameof(EOSManager)} ({nameof(s_lazyManagerInstance)}): Lazy loaded singleton instance is creating a GameObject with an EOSManager attached.");
+
+            // EOSManager is a MonoBehaviour, so create a GameObject to hold it
+            GameObject instanceGameObjectHolder = new GameObject();
+
+            // Add the component
+            EOSManager addedManager = instanceGameObjectHolder.AddComponent<EOSManager>();
+
+            return addedManager;
+        });
```

Comment:
A few thoughts on this:

* The `Lazy<T>` field member when used as the instance for a singleton class should use the `readonly` modifier. This is because the "Lazy Object" never changes, but _the thing it encapsulates_ does.
* If the intent is to utilize a singleton pattern, the constructor should be a `private` function, not an inline lambda function like it is here.

Reply:
The goal for not being readonly was that the `set` part of the `ManagerInstance` needs to be able to set an instance, in case the user is continuing to use the Singletons prefab object that has this behaviour attached. If it's readonly, I can't do that trick, and I don't think that `Lazy` has other ways of setting an instance like so.

Noted on the suggestion about not using the lambda. So a static private method, then?